build-tools/webpack/extract-manifest-plugin.js

```diff
@@ -54,79 +54,76 @@ class ExtractManifestPlugin {
-								// directly in the runtime.
-								return match;
-							}
-
-							// `nameExpression` is an expression that assumes `chunkId` exists in the scope (like `({1:'chunk1', 2:'chunk2'}[chunkId])`).
-							// Transform it to a function and add it to the manifest.
-							manifestContent.push(
-								`${ globalManifest }.set('${ variable }', function(chunkId){return ${ nameExpression };});`
+				// For each variable in brackets, generate a map chunkId->value for that variable and store it in a global function
+				return filenameExpression.replace( variableRegex, function ( match, variable ) {
+					// If the variable has been already captured, don't try to generate the value again
+					if ( ! variablesProcessed.has( variable ) ) {
+						// Call webpack again to generate the chunkId->value map for that specific variable.
+						let nameExpression;
+						try {
+							nameExpression = compilation.getAssetPath(
+								`"[${ variable }]"`,
+								{
+									...options,
+									// Avoid infinite loops
+									skipExtractManifestPlugin: true,
+								},
+								assetInfo
 							);
-							variablesProcessed.add( variable );
+						} catch {
+							// Error trying to get webpack to give us the replacement map. Return the original string
+							// without any replacement and let wepback continue doing its work. The worst case secenario is that we
+							// don't replace this variable with a global function, and therefore the chunkId->value map is embeded
+							// directly in the runtime.
+							return match;
 						}
-						// Create an expression that calls the generated manifest function with the chunkId
-						return `${ quote }+${ globalManifest }.get('${ variable }')(chunkId)+${ quote }`;
-					} );
-				}
-			);
+
+						// `nameExpression` is an expression that assumes `chunkId` exists in the scope (like `({1:'chunk1', 2:'chunk2'}[chunkId])`).
+						// Transform it to a function and add it to the manifest.
+						manifestContent.push(
+							`${ globalManifest }.set('${ variable }', function(chunkId){return ${ nameExpression };});`
+						);
+						variablesProcessed.add( variable );
+					}
+					// Create an expression that calls the generated manifest function with the chunkId
+					return `${ quote }+${ globalManifest }.get('${ variable }')(chunkId)+${ quote }`;
+				} );
+			} );
 
 			// When generating the files for the `runtime` chunk, add a second file with the content of the manifest. This will reuse the filename
 			// template and hash for both files (eg: it will generate `runtime.<hash>.js` and `manifest.<hash>.js` and both <hash> will be the same).
 			//
-			// The logic has been copied from webpack/lib/JavascriptModulesPlugin.js
-			compilation.mainTemplate.hooks.renderManifest.tap( PLUGIN_NAME, ( result, options ) => {
+			// The logic has been copied from webpack/lib/javascript/JavascriptModulesPlugin.js
+			compilation.hooks.renderManifest.tap( PLUGIN_NAME, ( result, options ) => {
```

Comment:
https://webpack.js.org/blog/2020-10-10-webpack-5-release/#mainchunkmoduletemplate-deprecation

Hook moved to `Compilation`